SemiconductorTestLibrary.Extensions/tests/Unit/InstrumentAbstraction/DAQmx/AnalogInputTests.cs

```diff
@@ -158,19 +174,53 @@ public void ReadFiveAnalogWaveformSamplesFromOneChannel_ResultsContainExpectedDa
             Assert.Equal(5, results.ExtractSite(1)["AIPin"].SampleCount);
         }
 
-        [Theory]
-        [InlineData("AIPin")]
-        public void ReadAnalogWaveformSamplesFromTwoFilteredChannels_ResultsContainExpectedData(string pinName)
+        [Fact]
+        [Trait(nameof(HardwareConfiguration), nameof(HardwareConfiguration.Lungyuan))]
+        public void ReadAnalogWaveformSamplesFromTwoFilteredChannels_ResultsContainExpectedData()
         {
             var sessionManager = Initialize("DAQmxMultiChannelTests.pinmap");
-            var tasksBundle = sessionManager.DAQmx(pinName);
-            var filteredResult = tasksBundle.FilterBySite(new int[] { 1, 3 }).ReadAnalogWaveform(100);
-            var site1Data = filteredResult.GetValue(filteredResult.SiteNumbers[0], pinName).Samples.FirstOrDefault().Value;
-            var site2Data = filteredResult.GetValue(filteredResult.SiteNumbers[1], pinName).Samples.FirstOrDefault().Value;
-
-            Assert.True(0.10 > site1Data && 0.01 < site1Data);
-            Assert.True(0.30 > site2Data && 0.20 < site2Data);
-            Assert.Equal(2, filteredResult.SiteNumbers.Length);
+            InitializeAndClose.CreateDAQmxAOVoltageTasks(_tsmContext);
+            var aiTasksBundle = sessionManager.DAQmx("AIPin");
+            var aoTasksBundle = sessionManager.DAQmx("AOPin");
+            var site0Data = AnalogWaveform<double>.FromArray1D(new double[] { 0.5 });
+            var site1Data = AnalogWaveform<double>.FromArray1D(new double[] { 0.8 });
+            var data = new PinSiteData<AnalogWaveform<double>>(new Dictionary<string, IDictionary<int, AnalogWaveform<double>>> { { "AOPin", new Dictionary<int, AnalogWaveform<double>> { { 0, site0Data }, { 1, site1Data } } } });
+            ConfigureTimingSampleClockSettings(aiTasksBundle, aoTasksBundle);
+
+            aoTasksBundle.WriteAnalogWaveform(data, false);
+            aiTasksBundle.Start();
+            aoTasksBundle.Start();
+            // Wait for the generation to get completed
+            Thread.Sleep(100);
+            aoTasksBundle.Stop();
+            var filteredSite = aiTasksBundle.FilterBySite(1).ReadAnalogWaveform();
+            aiTasksBundle.Stop();
+            var filteredWaveformSamples = filteredSite.GetValue(filteredSite.SiteNumbers[0], "AIPin").Samples;
+            var maxValueOfFilteredSamples = filteredWaveformSamples.Max(filteredSample => filteredSample.Value);
+
+            AssertFilteredSample(maxValueOfFilteredSamples, aiTasksBundle);
+            InitializeAndClose.ClearDAQmxAOVoltageTasks(_tsmContext);
+        }
+
+        private void ConfigureTimingSampleClockSettings(DAQmxTasksBundle aiTaskBundle, DAQmxTasksBundle aoTaskBundle)
```

Comment:
```suggestion
        private void ConfigureTimingSampleClockSettings(DAQmxTasksBundle aiTasksBundle, DAQmxTasksBundle aoTasksBundle)
```
Let's be consistent with the naming

Reply:
Updated